core/cloudflow-akka-testkit/src/main/scala/cloudflow/akkastream/testkit/TestContext.scala

```diff
@@ -202,38 +198,27 @@ private[testkit] case class TestContext(
       }
     }
 
-  def streamletExecution: StreamletExecution = new StreamletExecution() {
-    val readyFuture            = readyPromise.future
-    def completed: Future[Dun] = completionFuture
-    def ready: Future[Dun]     = readyFuture
-    def stop(): Future[Dun]    = TestContext.this.stop()
-  }
+  private val execution                               = new StreamletExecutionImpl(this)
```

Comment:
`StreamletExecutionImpl` is shared by the real streamlet context and the test context.